remote-storage-managers/hdfs/src/main/java/org/apache/kafka/rsm/hdfs/LogSegmentDataHeader.java

```diff
@@ -0,0 +1,205 @@
+package org.apache.kafka.rsm.hdfs;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.kafka.common.errors.UnsupportedVersionException;
+import org.apache.kafka.common.log.remote.storage.LogSegmentData;
+
+import java.io.File;
+import java.nio.ByteBuffer;
+import java.util.EnumMap;
+import java.util.Objects;
+
+public class LogSegmentDataHeader {
+
+    public enum FileType {
+        // NOTE: DONT CHANGE THE ORDER OF THE FILE TYPES. Once a new file type is added, make sure to change the version
+        // and handle the SERDE for backward compatibility.
+        OFFSET_INDEX((byte) 0),
+        TIMESTAMP_INDEX((byte) 1),
+        LEADER_EPOCH_CHECKPOINT((byte) 2),
+        PRODUCER_SNAPSHOT((byte) 3),
+        TRANSACTION_INDEX((byte) 4),
+        SEGMENT((byte) 5);
+
+        private final byte id;
+
+        FileType(byte id) {
+            this.id = id;
+        }
+
+        public static FileType fromId(byte id) {
+            switch (id) {
+                case 0:
+                    return OFFSET_INDEX;
+                case 1:
+                    return TIMESTAMP_INDEX;
+                case 2:
+                    return LEADER_EPOCH_CHECKPOINT;
+                case 3:
+                    return PRODUCER_SNAPSHOT;
+                case 4:
+                    return TRANSACTION_INDEX;
+                case 5:
+                    return SEGMENT;
+                default:
+                    return null;
+            }
+        }
+    }
+
+    public static final Integer LENGTH = 25;
+    public static final byte CURRENT_VERSION = 0;
+
+    private byte version;
+    private final EnumMap<FileType, Integer> filePositions = new EnumMap<>(FileType.class);
+
+    private LogSegmentDataHeader() {
+    }
+
+    @VisibleForTesting
+    byte version() {
+        return version;
+    }
+
+    @VisibleForTesting
+    EnumMap<FileType, Integer> filePositions() {
+        return filePositions;
+    }
+
+    public DataPosition getDataPosition(final FileType fileType) {
+        final Integer position = filePositions.get(fileType);
+        switch (fileType) {
+            case OFFSET_INDEX:
+            case TIMESTAMP_INDEX:
+            case LEADER_EPOCH_CHECKPOINT:
+            case PRODUCER_SNAPSHOT:
+            case TRANSACTION_INDEX:
+                final FileType nextFileType = FileType.fromId((byte) (fileType.id+1));
+                final int nextFilePosition = filePositions.get(nextFileType);
+                final int length = nextFilePosition - position;
+                return new DataPosition(position, length);
+            case SEGMENT:
+                return new DataPosition(position, Integer.MAX_VALUE);
+            default:
+                return null;
```

Comment:
You may want to throw an `IllegalArgumentException` instead of returning null.

Reply:
Ack.